src/distributors/base-installer.ts

```diff
@@ -11,14 +11,17 @@ export abstract class JavaBase {
   protected version: semver.Range;
   protected architecture: string;
   protected javaPackage: string;
+  protected stable: boolean;
 
   constructor(protected distributor: string, installerOptions: JavaInstallerOptions) {
     this.http = new httpm.HttpClient('setup-java', undefined, {
       allowRetries: true,
       maxRetries: 3
     });
 
-    this.version = this.normalizeVersion(installerOptions.version);
+    ({ version: this.version, stable: this.stable } = this.normalizeVersion(
```

Comment:
Why it needs extra round brackets?

Reply:
It is required by JS if you use object {} on left side of =